src/components/LineChart/index.tsx

```diff
@@ -68,20 +70,25 @@ const LineChart = ({
     textColor: theme.colors.neutral.textWeak,
   }
 
-  const [selected, setState] = useState(
-    dataset.datasets.map(({ id }, index) => `${id}${index}`),
+  const [selected, setSelected] = useState(
+    dataset.datasets?.map(({ id }, index) => `${id}${index}`),
   )
 
-  const finalData = dataset.datasets.filter(
-    ({ id }, index) => selected.indexOf(`${id}${index}`) > -1,
+  const finalData = dataset.datasets?.filter(({ id }, index) =>
+    selected ? selected.indexOf(`${id}${index}`) > -1 : false,
   )
 
+  useEffect(() => {
+    if (selected !== undefined) return
```

Comment:
Can you add bracket ? more clear

Reply:
I definitely prefer this compact formatting, but if that's part of the coding style guideline, no problem! It can also be forced by an eslint rule btw!